src/linux.rs

```diff
@@ -60,24 +60,59 @@ impl Output {
 
 #[cfg(target_os = "linux")]
 pub(crate) fn get_disk_id() -> Result<String, HWIDError> {
-    let mut com = Command::new("sh");
-    com.arg("-c").arg("lsblk -f -J -o NAME,MOUNTPOINT,UUID");
+    let output = run_command("lsblk -f -J -o NAME,MOUNTPOINT,UUID")?;
 
-    let output = com.output()?;
-
-    let output_string = String::from_utf8(output.stdout)?;
+    let output_string = String::from_utf8(output.into())?;
     let parsed: Output = serde_json::from_str(output_string.as_str())?;
     let uuid = parsed.get_root()?;
     Ok(uuid)
 }
 
+#[cfg(target_os = "linux")]
+fn run_command(command: &str) -> Result<String, HWIDError> {
+    let mut cmd = Command::new("sh");
+    let cmd = cmd.arg("-c")
+        .arg(command);
+
+    let output = cmd.output()?;
+    if !cmd.status()?.success() {
+        return Err(HWIDError::new(
+            &format!("Failed to run command: {command}"),
+            &String::from_utf8(output.stderr.into())?,
+        ));
+    }
+
+    Ok(String::from_utf8(cmd.output()?.stdout)?)
+}
+
+#[cfg(target_os = "linux")]
+fn get_mac_addressof_interface(interface_name: &str) -> Result<String, HWIDError> {
+    run_command(&format!("cat /sys/class/net/{interface_name}/address"))
+}
+
 #[cfg(target_os = "linux")]
 pub(crate) fn get_mac_address() -> Result<String, HWIDError> {
-    let mut com = Command::new("sh");
-    com.arg("-c")
-        .arg("cat /sys/class/net/$(ip route show default | awk '/default/ {print $5}')/address");
-    let output = com.output()?;
-    Ok(String::from_utf8(output.stdout)?)
+    // First check for first cable internet interface connected on hardware,
+    // if the ethernet cable is not available we find the first wifi interface.
+    let first_cable_interface_names = [vec!["eth0".to_string()]
+        , (0..10)
+            .map(|x| format!("enp{x}s0"))
+            .collect::<Vec<String>>()
+        , vec!["wlan0".to_string()]
+        , (0..10)
+            .map(|x| format!("wlp{x}s0"))
+            .collect::<Vec<String>>()
+    ].concat();
```

Comment:
I believe we should follow [this](https://www.freedesktop.org/wiki/Software/systemd/PredictableNetworkInterfaceNames/) instead.

Reply:
Isn't that naming scheme being included in this same commit?

Reply:
Is already being covered, I'm just fetching the first Ethernet interface. I see no point to use others.

Reply:
Sorry, but I believe it is covering only a part of the convention. Note:
>    Names incorporating Firmware/BIOS provided index numbers for on-board devices (example: eno1)
    Names incorporating Firmware/BIOS provided PCI Express hotplug slot index numbers (example: ens1)
    Names incorporating physical/geographical location of the connector of the hardware (example: enp2s0)
    Names incorporating the interfaces's MAC address (example: enx78e7d1ea46da)
    Classic, unpredictable kernel-native ethX naming (example: eth0)
>
> By default, systemd v197 will now name interfaces following policy 1) if that information from the firmware is applicable and available, falling back to 2) if that information from the firmware is applicable and available, falling back to 3) if applicable, falling back to 5) in all other cases. Policy 4) is not used by default, but is available if the user chooses so.
>
> This combined policy is only applied as last resort. That means, if the system has biosdevname installed, it will take precedence. If the user has added udev rules which change the name of the kernel devices these will take precedence too. Also, any distribution specific naming schemes generally take precedence. 

Probably beyond the scope of this PR, but despite that naming convention, the user can choose anything he wants, so for reliability, the best would be to list the present ethernet devices instead, probably using ioctl directly.

Reply:
Update to add support to `eno` and `ens` names.